cloudsmith_cli/cli/commands/push.py

```diff
@@ -54,6 +58,82 @@ def validate_upload_file(ctx, opts, owner, repo, filepath, skip_errors):
     return md5_checksum
 
 
+def init_multipart_upload(opts, owner, repo, filepath):
+    """
+    Initialize a multipart upload.
+    Returns a tuple of (identifier, upload_url, upload_headers)
+    """
+
+    filename = os.path.basename(filepath)
+    resp = requests.post(
```

Comment:
Would favour not using requests directly, but instead using the cloudsmith API to perform the upload. Are you able to use the API bindings?